fineract-provider/src/main/java/org/apache/fineract/infrastructure/core/service/TomcatJdbcDataSourcePerTenantService.java

```diff
@@ -80,15 +86,28 @@ public DataSource retrieveDataSource() {
     // creates the tenant data source for the oltp and report database
     private DataSource createNewDataSourceFor(final FineractPlatformTenantConnection tenantConnectionObj) {
         String protocol = toProtocol(this.tenantDataSource);
-        String jdbcUrl = toJdbcUrl(protocol, tenantConnectionObj.getSchemaServer(), tenantConnectionObj.getSchemaServerPort(),
-                tenantConnectionObj.getSchemaName(), tenantConnectionObj.getSchemaConnectionParameters());
+        // Default properties for Writing
+        String schemaServer = tenantConnectionObj.getSchemaServer();
+        String schemaPort = tenantConnectionObj.getSchemaServerPort();
+        String schemaName = tenantConnectionObj.getSchemaName();
+        String schemaUsername = tenantConnectionObj.getSchemaUsername();
+        String schemaPassword = tenantConnectionObj.getSchemaPassword();
+        // Properties to ReadOnly case
+        if (this.fineractProperties.getMode().isReadOnlyMode()) {
```

Comment:
I know I say this all the time but this should be covered by unit tests.

Reply:
How we can run the ReadOnly instance mode type, remember that If the instance mode is ReadOnly, Liquibase is not executed in the startup time and then we'll have not an initial datatabase schema.